service/interpreter/outputCollector.go

```diff
@@ -7,16 +7,29 @@ type OutputCollector struct {
 }
 
 func NewOutputCollector(initOutputs []iwfidl.StateCompletionOutput) *OutputCollector {
+	filteredOutputs := []iwfidl.StateCompletionOutput{}
+
 	if initOutputs == nil {
-		initOutputs = []iwfidl.StateCompletionOutput{}
-	}
-	return &OutputCollector{
-		outputs: initOutputs,
+		return &OutputCollector{
+			outputs: filteredOutputs,
+		}
+	} else {
+		for _, output := range initOutputs {
+			if output.CompletedStateOutput != nil {
+				filteredOutputs = append(filteredOutputs, output)
+			}
+		}
+
+		return &OutputCollector{
+			outputs: filteredOutputs,
+		}
 	}
 }
```

Comment:
@longquanzheng is there a reason why we wouldn't filter out outputs with nil CompletedStateOutput here?

Reply:
Discussed